### PR TITLE
Core layers consistency

### PR DESCRIPTION
**Pull request by pmeijer**
First go on making a consistent style on the core layers.

The inner core passed should be named innerCore, the passed on core should be named core.

Private helper functions should be function declarations at the top (Helper Functions).

Overwritten functions should be "Modified Methods" and new/added functions should be "Added Methods". This distinction is partially done (there is no longer any intermediate local variables at least).

### CODE REVIEW

File: src/common/core/constants.js [hunk 1]
```diff
@@ -12,6 +12,35 @@ define([], function () {
...
+
+        SET_ITEMS: 'items',
+        SET_ITEMS_MAX: 'max',
+        SET_ITEMS_MIN: 'min',
+
+        META_ASPECTS: 'aspects',
+        META_CHILDREN: 'children',
+        META_NODE: '_meta',
+        META_POINTER_PREFIX: '_p_',
+        META_ATTRIBUTE_PREFIX: '_a_',
```
Comment (kecso) [inline src/common/core/constants.js]: sorry I was wrong, this prefix is used for ASPECTS

File: src/common/core/coretree.js [hunk 2]
```diff
@@ -24,23 +24,17 @@ define([
...
-
+            logger = options.logger.fork('core'),
             ID_NAME = storage.ID_NAME,
-            __getEmptyData = function () {
-                return {};
-            },
             roots = [],
             ticks = 0,
+            mutateCount = 0,
+            checkValidTreeRunning = true,
```
Comment (kecso) [inline src/common/core/coretree.js]: this variable seems to be out of use, we should probably remove it

File: src/common/core/metacachecore.js [hunk 3]
```diff
@@ -132,28 +153,22 @@ define(['common/util/assert', 'common/core/tasync'], function (ASSERT, TASYNC) {
...
-                }
-            }
-
-            //parameters
-            // node - the node in question
-            // children - the current children of the node, so that multiplicity can be checked
-            // sensitive - if true the function do not return the connection and abstract types
-            // multiplicity - if true the function filters out possibilities that fail multiplicity check
-            // aspect - if given the function also filters out valid children type meta nodes based on aspect rule
+            /**
```
Comment (kecso) [inline src/common/core/metacachecore.js]: this comment can be removed as the function description is done in the core.js

File: src/common/core/constants.js (hunk 1 above)
Comment (pmeijer) [inline src/common/core/constants.js]: META_ASPECT_PREFIX then?

Comment (kecso) [inline src/common/core/constants.js]: yes, that would be perfect

**Pull request merged**